biospecimen-tracking/api/src/main/java/org/phenotips/data/internal/controller/BiospecimensController.java

```diff
@@ -0,0 +1,379 @@
+     * @return a EntityReference object
+     */
+    public static EntityReference getClassReference()
+    {
+        return new EntityReference(CLASS_NAME, EntityType.DOCUMENT, Constants.CODE_SPACE_REFERENCE);
+    }
+
+    /***************************************** Private Methods ****************************************/
+
+    /**
+     * Helper function. If selectedFieldNames is no null it will be returned, otherwise the PROPERTIES set will be
+     * returned otherwise.
+     *
+     * @param selectedFieldNames a collection of field names to return if not null
+     * @return a Collection
+     */
+    private static Collection<String> propertyNamesToIterateOver(Collection<String> selectedFieldNames)
+    {
+        if (CollectionUtils.isEmpty(selectedFieldNames)) {
+            return BiospecimenData.PROPERTIES;
+        } else {
+            return selectedFieldNames;
+        }
+    }
+
+    private static BiospecimenData parseBiospecimenObject(Object biospecimenObject, DateFormat jsonDateFormat)
+        throws ParseException {
+
+
+        if (!(biospecimenObject instanceof JSONObject)) {
+            return null;
+        }
+
+        JSONObject biospecimenJSONObject = (JSONObject) biospecimenObject;
+
+        BiospecimenData biospecimen = new BiospecimenData();
+
+        for (String property : BiospecimenData.PROPERTIES) {
+            if (!biospecimenJSONObject.has(property)) {
+                continue;
+            }
+
+            Object propertyObject = biospecimenJSONObject.get(property);
+
+            if (propertyObject == null) {
+                continue;
+            }
+
+            switch (property) {
+                case BiospecimenData.TYPE_PROPERTY_NAME:
+                    biospecimen.setType(String.valueOf(propertyObject));
+                    break;
+                case BiospecimenData.DATE_COLLECTED_PROPERTY_NAME:
+                    biospecimen.setDateCollected(jsonDateFormat.parse(String.valueOf(propertyObject)));
+                    break;
+                case BiospecimenData.DATE_RECEIVED_PROPERTY_NAME:
+                    biospecimen.setDateReceived(jsonDateFormat.parse(String.valueOf(propertyObject)));
+                    break;
+                default:
+                    break;
+            }
+        }
+
+        return biospecimen;
+    }
+
+    private static JSONObject getJSONObjectFromBiospecimenData(BiospecimenData biospecimen,
+        Collection<String> propertiesToInclude, DateFormat jsonDateFormat)
+    {
+        Collection<String> propertiesToIterateOver = propertiesToInclude;
+        if (CollectionUtils.isEmpty(propertiesToIterateOver)) {
+            propertiesToIterateOver = BiospecimenData.PROPERTIES;
+        }
+
+        JSONObject biospecimenObject = new JSONObject();
+
+        boolean jsonIsEmpty = true;
+        for (String propertyName : propertiesToIterateOver) {
+
```

Comment:
Again, I don't think the `for` + `switch` is necessary here, since you're explicitly handling each property. I don't see any reason not to just pull out the `if`s inside each `case` and then remove the `for` + `switch`.